search-app/src/cmr/search/services/acls/collections_cache.clj

```diff
@@ -116,62 +116,35 @@
-  (let [coll-by-provider-id-and-entry-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key)
+  "Gets a single collection from the cache by concept id. If collection is not found, will add it to the cache (if it exists in elastic) and will return found collection."
+  (let [coll-by-provider-id-and-entry-title-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key)
+       _ (info (str "coll cache = " (pr-str coll-by-provider-id-and-entry-title-cache)))
         collection (hash-cache/get-value
-                    coll-by-provider-id-and-entry-id-cache ;; cache
-                    coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key ;; key
-                    (str provider-id entry-title))]
-   (when (or (nil? collection) (empty? collection))
-    (info (coll-msg/collection-not-found (str provider-id entry-title)))
-    (coll-for-gran-acl-caches/refresh-entire-cache context)) ;;TODO replace
-   (time-strs->clj-times collection))))
-
-;; TODO remove after transition
-(defn- get-collections-map
-  "Gets the cached value. By default an error is thrown if there is no value in
-   the cache. Pass in anything other then :return-errors for the third parameter
-   to have null returned when no data is in the cache. :return-errors is for the
-   default behavior."
-  ([context key-type]
-   (get-collections-map context key-type :return-errors))
-  ([context key-type throw-errors]
-  (let [coll-cache (hash-cache/context->cache context cache-key)
-        collection-map (hash-cache/get-value
-                        coll-cache
-                        cache-key
-                        key-type)]
-    (if (and (= :return-errors throw-errors) (empty? collection-map))
-      (errors/internal-error! (coll-msg/collections-not-in-cache key-type))
-      (time-strs->clj-times collection-map)))))
-
-;; TODO remove after transition
-(defn get-collection
-  "Gets a single collection from the cache by concept id. Handles refreshing the cache if it is not found in it.
-  Also allows provider-id and entry-title to be used."
-  ([context concept-id]
-   (let [by-concept-id (get-collections-map context :by-concept-id :no-errors)]
-     (when-not (and by-concept-id (by-concept-id concept-id))
-       (info (coll-msg/collection-not-found concept-id))
-       (refresh-cache context))
-     (get (get-collections-map context :by-concept-id ) concept-id)))
-  ([context provider-id entry-title]
-   (let [by-provider-id-entry-title (get-collections-map context :by-provider-id-entry-title )]
-     (get by-provider-id-entry-title [provider-id entry-title]))))
+                   coll-by-provider-id-and-entry-title-cache
+                   coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key
+                   (str provider-id entry-title))]
+  (if (or (nil? collection) (empty? collection))
```

Comment:
(empty? nil) returns true, feels like you could simplify?

Reply:
This logic existed before, so I assumed there might be a case where collection is in cache, but it is empty for some reason, so I kept it in there just in case